PWGCF/Flow/Tasks/flowMc.cxx

```diff
@@ -0,0 +1,323 @@
+
+    histos.add<TH1>("hPhi", "#phi distribution", HistType::kTH1D, {axisPhi});
+    histos.add<TH1>("hPhiWeighted", "corrected #phi distribution", HistType::kTH1D, {axisPhi});
+
+    if (cfgOutputNUAWeights) {
+      o2::framework::AxisSpec axis = axisPt;
+      int nPtBins = axis.binEdges.size() - 1;
+      double* ptBins = &(axis.binEdges)[0];
+
+      fWeights->setPtBins(nPtBins, ptBins);
+      fWeights->init(true, false);
+    }
+  }
+
+  void loadCorrections(uint64_t timestamp)
+  {
+    if (correctionsLoaded)
+      return;
+    if (cfgFlowAcceptance.value.empty() == false) {
+      mAcceptance = ccdb->getForTimeStamp<GFWWeights>(cfgFlowAcceptance, timestamp);
+      if (mAcceptance)
+        LOGF(info, "Loaded acceptance weights from %s (%p)", cfgFlowAcceptance.value.c_str(), (void*)mAcceptance);
+      else
+        LOGF(warning, "Could not load acceptance weights from %s (%p)", cfgFlowAcceptance.value.c_str(), (void*)mAcceptance);
+    }
+    if (cfgFlowEfficiency.value.empty() == false) {
+      mEfficiency = ccdb->getForTimeStamp<TH1D>(cfgFlowEfficiency, timestamp);
+      if (mEfficiency == nullptr) {
+        LOGF(fatal, "Could not load efficiency histogram for trigger particles from %s", cfgFlowEfficiency.value.c_str());
+      }
+      LOGF(info, "Loaded efficiency histogram from %s (%p)", cfgFlowEfficiency.value.c_str(), (void*)mEfficiency);
+    }
+    correctionsLoaded = true;
+  }
+
+  bool setCurrentParticleWeights(float& weight_nue, float& weight_nua, float phi, float eta, float pt, float vtxz)
+  {
+    float eff = 1.;
+    if (mEfficiency)
+      eff = mEfficiency->GetBinContent(mEfficiency->FindBin(pt));
+    else
+      eff = 1.0;
+    if (eff == 0)
+      return false;
+    weight_nue = 1. / eff;
+    if (mAcceptance)
+      weight_nua = mAcceptance->getNUA(phi, eta, vtxz);
+    else
+      weight_nua = 1;
+    return true;
+  }
+
+  using RecoTracks = soa::Join<aod::TracksIU, aod::TracksExtra>;
+
+  void process(aod::McCollision const& mcCollision, soa::Join<aod::McParticles, aod::ParticlesToTracks> const& mcParticles, RecoTracks const&)
+  {
+
+    float imp = mcCollision.impactParameter();
+    float evPhi = mcCollision.eventPlaneAngle();
+    float vtxz = mcCollision.posZ();
+    if (evPhi < 0)
+      evPhi += constants::math::TwoPI;
+
+    int64_t nCh = 0;
+    float weff = 1.;
+    float wacc = 1.;
+    auto bc = mcCollision.bc_as<aod::BCsWithTimestamps>();
+    loadCorrections(bc.timestamp());
+
+    if (imp > minB && imp < maxB) {
+      // event within range
+      histos.fill(HIST("hImpactParameter"), imp);
+      histos.fill(HIST("hEventPlaneAngle"), evPhi);
+
+      for (auto const& mcParticle : mcParticles) {
+        // focus on bulk: e, mu, pi, k, p
+        int pdgCode = std::abs(mcParticle.pdgCode());
+        if (pdgCode != 11 && pdgCode != 13 && pdgCode != 211 && pdgCode != 321 && pdgCode != 2212)
```

Comment:
Please, for the next iteration use named constants from `TPDGCode.h` instead of numbers